### ISSUE TITLE
Getting wrong position value when using MediaSessionCompat.Callback.onSeekTo() with MediaSessionConnector

### ISSUE THREAD

**Issue by nandovelazquez** [google/ExoPlayer#8333]
Implementing the `MediaSession` and `MediaSessionConnector` on the ExoPlayer Demo and dictating the `fast-forward` command through Google Assistant, the returned value by `MediaSessionCompat.Callback.onSeekTo()` does not seem to be correct, it looks like it includes extra time.

I'm using the latest ExoPlayer version(2.12.2) and following the [MediaSessionExtension lab](https://developer.android.com/codelabs/supporting-mediasession#0)

Hope you can guide me if I need to set an extra setting or something and sorry if this is an already answered question but I could not find any info related.

Thanks!

**Comment by marcbaechinger** [comment]
I understand you are dictating "Ok Google, fast forward" and this triggers the `MediaSessionCompat.Callback.onSeekTo()` instead of `MediaSessionCompat.Callback.onFastForward`. If that is the intended behaviour then the client (like Google Assistant that is processing the voice command) is calculating `currentPosition + fastFowardIncrement` and send this value by using `TransportControls.seekTo(positionMs)`. The `MediaSessionConnector` then [receives this value](https://github.com/google/ExoPlayer/blob/release-v2/extensions/mediasession/src/main/java/com/google/android/exoplayer2/ext/mediasession/MediaSessionConnector.java#L1170) and  executes the seek.

Can you set a break point int the `MediaSessionConnector` to check what values arrives and what you would expect? From what you report it seems that the value that arrives is too large (you say 'extra time is added'). 

Is the `currentPosition` that Assistant is maintaining after having it received from the `PlaybackStateCompat` correct?  Is this displayed somewhere in a UI?

What is the expected `fastForwardIncrement` for the voice call you are issuing and what is what you get when [putting a break point in the callback](https://github.com/google/ExoPlayer/blob/release-v2/extensions/mediasession/src/main/java/com/google/android/exoplayer2/ext/mediasession/MediaSessionConnector.java#L1170)?

**Comment by nandovelazquez** [comment]
Hi @marcbaechinger, thanks for your help.

Steps to reproduce it:
- Open the ExoPlayer
- From the **Clear DASH** menu, select the video **HD(MP4, H264)**
- Being the playhead position in 0, dictate the voice command **"fast forward one minute"** through google assistant.
The expectation is that the new position should be 60000 and the playhead at the 1 minute on the progress bar.

Having the breakpoint where you pointed out, the `positionMs` value varies every time, some examples: 
**72837**, **68121**, **65669**, **68164**, and on the progress bar the playhead appears at **1.13** minute, **1.08**, **1.06**, **1.08**, respectively. 

As you can see every time the new position gets extra time and the time I get on the `onSeek()` method in millis doesn't correspond to what I see on the progress bar.

Let me know if you need more info.

**Comment by marcbaechinger** [comment]
I was able to reproduce this behaviour with the demo app. I get the same data points like you get which means that Assistant seems to send the new position calculated wrongly. I can't see an apparent misbehavior on the `MediaSessionConnector/ExoPlayer` side though. Here is what I've done

1. Start playback in a foreground service.
2. Pause playback and seek to position 0
3. Run `adb shell dumpsys media_sesssion` which give the following output:

```
audio_demo com.google.android.exoplayer2.demo/audio_demo (userId=0)
      ownerPid=29226, ownerUid=10178, userId=0
      package=com.google.android.exoplayer2.demo
      launchIntent=null
      mediaButtonReceiver=PendingIntent{686014e: PendingIntentRecord{b6eba60 com.google.android.exoplayer2.demo broadcastIntent}}
      active=true
      flags=3
      rating type=0
      controllers: 4
      state=PlaybackState {state=2, position=0, buffered position=2638, speed=0.0, updated=3104460, actions=6554447, custom actions=[], active item id=0, error=null}
      audioAttrs=AudioAttributes: usage=USAGE_MEDIA content=CONTENT_TYPE_UNKNOWN flags=0x800 tags= bundle=null
      volumeType=1, controlType=2, max=0, current=0
      metadata: size=7, description=Jazz in Paris, null, Jazz for the masses
      queueTitle=null, size=1
```

The important bits are 'state=2, position=0, buffered position=2638, speed=0.0' which means that the state of the media session is 2 (=paused), the position is 0 and the speed is 0.0 as it needs to be when in paused state. This verifies that the session reflects the state of the player correctly regarding position and state.

4. Issue voice command: "Fast forward one minute"
5. The position that arrives at 'MediaSessionCompat.Callback.onSeekTo(long positionMs)' is always a few seconds above the expected 60 seconds. It seems that it just adds the time it takes me to issue the voice command, which is usually a few seconds. When I wait for about 25 seconds with the voice command, the position sent to the connector is about 25 seconds higher.

The expected behavior would be that playback continues at 60 seconds and the `positionMs` that arrives at the `seekTo` method of the callback is expected to be 60000. 

It seems that Assistant does calculate the position wrongly, something like:

`position + timeSinceLastPlaybackStateUpdate + fastForwardTime`

When the player is never paused since the playback state update, this works correctly. But when paused for a while the duration of pausing is added to the position sent to the connector.

**Comment by ojw28** [comment]
[Internal ref: b/176093999]

**Comment by nandovelazquez** [comment]
Thanks for your help @marcbaechinger. 
Is this going to be addressed to the Google Assistant team? If so, can I get a reference to follow the status of the new ticket?

**Comment by marcbaechinger** [comment]
The bug is filed internally and can be tracked internally only I'm afraid. We will update this issue if there is some news.

**Comment by craigrbruce** [comment]
Hi. I have heard from an internal contact at Google that this issue has been fixed and released. Could you please confirm here if that is the case? Thanks.

**Comment by nandovelazquez** [comment]
I'd like to know about that release too 👀

Was it released in a patch? What Android versions could get this patch?

**Comment by marcbaechinger** [comment]
Thanks for asking. I'm glad to have someone who can verify if this landed on devices and whether the fix helps.

This is fixed in the source code of the search app. I'll try to get the version of the search app that includes the fix and let you know as soon as I know more. With the version you can check whether the device you are using has the fix already.

**Comment by bcdarius** [comment]
@marcbaechinger Do you have any further updates here? Thanks

**Comment by marcbaechinger** [comment]
@bcdarius Can you let me know with which Google Search App on Android you tested with on which you see the problem? It should be fixed since version 12.18.6.

**Comment by ojw28** [comment]
I think we can close this as fixed. However, please respond here if you still see the issue when using versions of the Google Search App that should have included the fix (i.e., >= 12.18.6).
